source/extensions/tracers/skywalking/trace_segment_reporter.cc

```diff
@@ -0,0 +1,165 @@
+#include "extensions/tracers/skywalking/trace_segment_reporter.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace SkyWalking {
+
+namespace {
+static constexpr uint32_t DEFAULT_DELAYED_SEGMENTS_CACHE_SIZE = 1024;
+
+// Convert SegmentContext to SegmentObject.
+TraceSegmentPtr toSegmentObject(const SegmentContext& segment_context) {
+  auto new_segment = std::make_unique<SegmentObject>();
+  SegmentObject& segment_object = *new_segment;
+
+  segment_object.set_traceid(segment_context.traceId());
+  segment_object.set_tracesegmentid(segment_context.traceSegmentId());
+  segment_object.set_service(segment_context.service());
+  segment_object.set_serviceinstance(segment_context.serviceInstance());
+
+  for (const auto& span_store : segment_context.spanList()) {
+    if (!span_store->sampled()) {
+      continue;
+    }
+    auto* span = segment_object.mutable_spans()->Add();
+
+    span->set_spanlayer(SpanLayer::Http);
+    span->set_spantype(span_store->isEntrySpan() ? SpanType::Entry : SpanType::Exit);
+    span->set_componentid(6000);
```

Comment:
Let's add a TODO here so we can set the correct component id here. cc. @wu-sheng

Reply:
Submit a pull request to these files, 
- https://github.com/apache/skywalking/blob/master/oap-server/server-bootstrap/src/main/resources/component-libraries.yml#L479.
- doc, https://github.com/apache/skywalking/blob/master/docs/en/guides/Component-library-settings.md#id-rules

And C++ should take [9000, 10000)